src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -420,25 +420,30 @@ HeapWord* G1CollectedHeap::attempt_allocation_slow(size_t word_size) {
   HeapWord* result = NULL;
   for (uint try_count = 1, gclocker_retry_count = 0; /* we'll return */; try_count += 1) {
     bool should_try_gc;
+    bool force_gc = false;
```

Comment:
`force_gc` and `should_try_gc` seems to overlap a bit here. At least the naming isn't perfect because we may not do a gc even if `force_gc` is true which I'd kind of expect.

I do not have a good new name right now how to fix this.

Reply:
It will be removed as part of my next round of changes